mm2src/kdf_walletconnect/src/connection_handler.rs

```diff
@@ -0,0 +1,137 @@
+use crate::storage::WalletConnectStorageOps;
+use crate::WalletConnectCtxImpl;
+
+use common::executor::Timer;
+use common::log::{debug, error, info};
+use futures::channel::mpsc::{UnboundedReceiver, UnboundedSender};
+use futures::StreamExt;
+use relay_client::error::ClientError;
+use relay_client::websocket::{CloseFrame, ConnectionHandler, PublishedMessage};
+use std::sync::Arc;
+
+const INITIAL_RETRY_SECS: f64 = 5.0;
+const MAX_BACKOFF: u64 = 60;
+const RETRY_INCREMENT: f64 = 5.0;
+
+pub struct Handler {
+    name: &'static str,
+    msg_sender: UnboundedSender<PublishedMessage>,
+    conn_live_sender: UnboundedSender<Option<String>>,
+}
+
+impl Handler {
+    pub fn new(
+        name: &'static str,
+        msg_sender: UnboundedSender<PublishedMessage>,
+        conn_live_sender: UnboundedSender<Option<String>>,
+    ) -> Self {
+        Self {
+            name,
+            msg_sender,
+            conn_live_sender,
+        }
+    }
+}
+
+impl ConnectionHandler for Handler {
```

Comment:
In next PRs, after the event streaming PR is merged we should start thinking about walletconnect events sent to the GUI. Please add this to the issue checklist.